internal/app/upstream/client.go

```diff
@@ -51,6 +51,11 @@ type Client interface {
 	// All goroutines that depend on the ctx object should consider ctx.Done to be related to shutdown.
 	// All such scenarios need to exit cleanly and are not considered an erroneous situation.
 	OpenStream(transport.Request, string) (<-chan transport.Response, func())
+
+	// GetStream is used as a debugging tool to get the latest state of the stream for an aggregated key
+	// If the aggregated key is absent, then either the stream was not created or was already cancelled.
+	// If present, returns the current version of the response from upstream
+	GetStream(string) (string, error)
```

Comment:
Is the version enough or would this be more useful for debugging if we returned the entire response? `GetStreamLatestVersion` might be more accurate.

Reply:
the version is enough IMO to know the latest on the stream. Will change the name.